lib/src/foundation/bloc.dart

```diff
@@ -3,6 +3,21 @@ import 'dart:async';
 import 'package:flutter/widgets.dart';
 import 'package:provider/provider.dart';
 
+class BlocProvider extends StatelessWidget {
```

Comment:
Why?

Reply:
Because we will be sure that the `dispose` method is called as now this responsibility is only inside the bloc.dart file